kaldb/src/main/java/com/slack/kaldb/logstore/search/SearchResultUtils.java

```diff
@@ -390,6 +391,72 @@ public static SearchResult<LogMessage> fromSearchResultProto(
             protoSearchResult.getInternalAggregations().toByteArray()));
   }
 
+  public static FieldType fromSchemaDefinitionProto(
+      KaldbSearch.SchemaDefinition protoSchemaDefinition) {
+    if (protoSchemaDefinition.getType().equals(KaldbSearch.FieldType.TEXT)) {
+      return FieldType.TEXT;
+    } else if (protoSchemaDefinition.getType().equals(KaldbSearch.FieldType.STRING)) {
+      return FieldType.STRING;
```

Comment:
does this need to account for string being defined as `keyword` by Opensearch/ES?

Reply:
This just handles serialization/deserialization of our defined `FieldType` values, which does not include a keyword type. I'm not sure we even need string/text differentiation as far as what we return out to the user, but for completeness of serialization support they're both include here.

As mentioned above, I think we probably need another field abstraction as what we return out but that will require additional schema work.